zebra/zebra_rib.c

```diff
@@ -1047,6 +1050,7 @@ void rib_install_kernel(struct route_node *rn, struct route_entry *re,
 	srcdest_rnode_prefixes(rn, &p, &src_p);
 
 	if (info->safi != SAFI_UNICAST) {
+		SET_FLAG(re->status, ROUTE_ENTRY_INSTALLED);
```

Comment:
A general, possibly dumb question, why do we auto-set the install/uninstall flag for routes if they are SAFI_UNICAST? I saw we do the same thing in `rib_uninstall_kernel()`.

Reply:
It's a bit the other way 'round: we directly set the flags for the _multicast_ family, because we're not actually going through the kernel/fib installation path for them. I added this line because we're sort of migrating to prefer use of the route_entry level flag as the shared nexthop-group concept takes over, and I noticed this path doesn't set that re flag.